helm/thingsboard/templates/component-service.yaml

```diff
@@ -32,6 +32,12 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
+  {{- if $componentValues.service.ipv6}}
+  ipFamilies:
+    - IPv6
```

Comment:
Why is this so different from EVP1? Can we remove the rest of the code that was there?

Reply:
What do you mean?

Reply:
I see that [here](https://github.com/midokura/thingsboard-private/blob/staging/k8s/charts/thingsboard/templates/tb-mqtt-transport.yaml) we do some if thens both for ipv4 and ipv6. So it could be that by setting ipFaimilies to IPv6 only we don't support ipv4 which might be undesired.

Does current PR creates MQTT that supports both ipv4 and ipv6 when  service.ipv6 is true?

Reply:
> So it could be that by setting ipFaimilies to IPv6 only we don't support ipv4 which might be undesired.

No, I have tested locally and it works

> Does current PR creates MQTT that supports both ipv4 and ipv6 when service.ipv6 is true?

That's the idea. Although I could not make it work with v6 locally. I tried to debug it with Xavi but we couldn't.

If I had to bet, I would say that my PR and the one linked in the ticket don't work out of the box unless there is some other stuff configured for having dual stack in the cluster. To fix it, someone added the code you linked, which creates two different services, one for v4 and another for v6, instead of using only one service with dual stack. And they left the other if branch which creates only one service in the case that only v4 or v6 are enabled, and in the later case it probably doesn't work.

Reply:
Hmmm, I got lost. Does current PR work or not? If it does not, why don't we follow the EVP1 code with different services for ipv4 and ipv6?

Reply:
We can't test it anyway, we don't have IPv6 devices. I have tried both approaches and neither of them works for me locally. After discussion with Xavi we have decided to merge this, and revisit it again if the requirement ever comes when we operate EVP2. But for now this is just about feature parity with EVP1, and we are not even sure whether it's actually needed, so it doesn't make sense to spend much time here.